drivers/overlay/overlay.go

```diff
@@ -2020,11 +2021,24 @@ func (d *Driver) getWhiteoutFormat() archive.WhiteoutFormat {
 }
 
 type overlayFileGetter struct {
-	diffDirs []string
+	diffDirs        []string
+	composefsMounts map[string]*os.File // map from diff dir to the directory with the composefs blob mounted
 }
 
 func (g *overlayFileGetter) Get(path string) (io.ReadCloser, error) {
+	buf := make([]byte, unix.PathMax)
 	for _, d := range g.diffDirs {
+		if f, found := g.composefsMounts[d]; found {
+			// there is no *at equivalent for getxattr, but it can be emulated by opening the file under /proc/self/fd/$FD/$PATH
+			len, err := unix.Getxattr(fmt.Sprintf("/proc/self/fd/%d/%s", int(f.Fd()), path), "trusted.overlay.redirect", buf)
+			if err != nil {
+				continue
```

Comment:
Should this be logrus.Error if not ENOENT?

Reply:
I've not investigated it further, but from the man page it seems `ENODATA` is returned:

```
       ENODATA
              The  named  attribute  does not exist, or the process has no access to this attribute.
```

other failures seem more difficult to hit, but if you prefer I can return the error to the caller if it is not `ENODATA`

Reply:
changed now to ignore only `ENODATA`

Reply:
sgtm